src/Matcher.js

```diff
@@ -125,10 +125,18 @@ export default class Matcher {
       params[name] = value && decodeURIComponent(value);
     });
 
-    return {
-      params,
-      remaining: pathname.slice(match[0].length),
-    };
+    let remaining = pathname.slice(match[0].length);
+    if (this.endsWithDelimiter(match[0])) {
```

Comment:
what is the overall question here? It seems like it'd be really weird for the child to match on `/foo` no?